### PR TITLE
Enhancement/improve e2e usage

### PR DESCRIPTION
**Pull request by sanderegg**
## What do these changes do?

Running the e2e as is done in the CI is cumbersome locally.
Much of the complexity of the e2e bash scripts were moved into the Makefile and optimized (less changes in host system are necessary)

The CI does:
```console
cd tests/e2e
make install-ci
make test
make clean-up
```
The developer can do:
```console
cd tests/e2e
make install-prod # or -ci if she/he wants to do the exact same thing
make test
make clean-up
```

The Makefile installation recipe does in order:
- pull the simcore stack from Dockerhub (if install-ci is used, it will try to find the branch name)
- build the simcore stack (if install-ci is used, this is a fallback)
- modify the host file/docker engine to allow usage of a local registry
- start a local registry
- setup python/node environment for puppeteer/python test infrastructure
- wait for the stack to be up
- transfer the sleeper 1.0.0 from Dockerhub to the local registry
- inject the sleepers template into the local postgres DB

The Makefile test procedure:
- runs the small tests
- runs the sleepers tutorial

The Makefile clean-up procedure:
- leaves the swarm (thus killing all services)
- stops the local registry
- cleans the host file and the docker engine configuration

## How to test

```console
# ensure the host is clean
make clean-all
# go to e2e
cd tests/e2e
make install-prod
make test
```

**Note for people testing 3rd party services locally:**
- the main Makefile allow to do ```make local-registry``` which will setup the simcore stack to use a local registry in ```registry:5000``` and you can push whatever service there (e.g. ```registry:5000/simcore/services/dynamic/my/awesom/service/that/fix/everything:2.3.4```)

## Checklist

- [ ] Did you change any service's API? Then make sure to bundle document and upgrade version (``make openapi-specs``, ``git commit ...`` and then ``make version-*``)
- [ ] Unit tests for the changes exist
- [ ] Runs in the swarm
- [ ] Documentation reflects the changes
- [ ] New module? Add your github username to [.github/CODEOWNERS](.github/CODEOWNERS)

### CODE REVIEW

**Review by GitHK — APPROVED** [review]
This looks very good. Easy to run locally!

**Review by ignapas — APPROVED** [review]

File: tests/e2e/Makefile [hunk 1]
```diff
@@ -1,72 +1,75 @@
...
-		--host 127.0.0.1 \
-		--port $(PUBLISHED_PORT) \
-		--user scu \
-		--dbname simcoredb \
-		--command "\copy projects from 'tutorials/sleepers_project_template_sql.csv' csv header;";
-
-
+include ../../scripts/common.Makefile
+
+SIMCORE_DOT_ENV = $(abspath $(CURDIR)/../../.env)
```
Comment (pcrespov) [inline tests/e2e/Makefile:3]: FYI: ``common.Makefile`` adds ``$(REPO_BASE_DIR)``

File: tests/e2e/Makefile [hunk 2]
```diff
@@ -1,72 +1,75 @@
...
+# inject the sleepers template
+@docker cp tutorials/sleepers_project_template_sql.csv $$(docker ps -q --filter="name=postgres"):/template_projects_sleepers_project_template_sql.csv
+# template is copied into the postgres container and then injected
+@docker exec $$(docker ps -q --filter="name=postgres") psql --user scu --dbname simcoredb --command "\copy projects from '/template_projects_sleepers_project_template_sql.csv' csv header;"
+endef
+
+# SETUP ##################
+
+.PHONY: install-ci install-prod clean-up
+install-ci install-prod: ## install e2e testing framework and start simcore/registry
```
Comment (pcrespov) [inline tests/e2e/Makefile:51]: Q? Why not to include also the ``devel`` stack?

**Review by pcrespov — APPROVED** [review]
Just some minors...
Thx! This makes everybody's work more easy! 🙏

File: tests/e2e/Makefile (hunk 2 above)
Comment (sanderegg) [reply tests/e2e/Makefile:51]: the devel stack is a pain. it blocks the process, so I tried to start it in a separate process but then it's a lot fiddling around to get a correct timing. I decided to ditch it.

**Pull request merged**